aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2274,6 +2285,41 @@ impl AptosVM {
             },
         })
     }
+
+    pub fn get_required_deposit(
+        &self,
+        session: &mut SessionExt,
+        resolver: &impl AptosMoveResolver,
+        txn_gas_params: &TransactionGasParameters,
+        txn_metadata: &TransactionMetadata,
+        payload: &TransactionPayload,
+    ) -> Option<u64> {
+        match payload {
+            TransactionPayload::EntryFunction(entry_func) => {
+                if self.randomness_enabled
+                    && has_randomness_attribute(resolver, session, entry_func).unwrap_or(false)
```

Comment:
This has to be `has_randomness_attribute(resolver, session, entry_func)?` and the return type of `get_required_deposit` should be a result of an option. Here you are remapping storage error to non-existent attribute which is wrong

Reply:
fixed

Reply:
to me it means we do have to tolerate those `get_required_deposit` errors?